File: medium/56/merge_intervals.py

class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        
        if len(intervals) == 0:
            return intervals

        intervals.sort()
        out = [intervals[0]]

        for interval in intervals:
            
            if out[-1][1] >= interval[0]:
                out[-1][1] = max(interval[1], out[-1][1])
            else:
                out.append(interval)
                
        return out

File: medium/56/merge_intervals.cpp

#include <algorithm>

class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {

        if (intervals.size() == 0) {
            return intervals;
        }
        
        int p = 0;
        vector<vector<int>> out;
        
        std::sort(intervals.begin(), intervals.end());
        out.push_back(intervals[0]);

        for (auto interval : intervals) {            
            if (out[p][1] >= interval[0]) {
                out[p][1] = std::max(out[p][1], interval[1]);
            }
            else {
                out.push_back(interval);
                p++;
            }
        }
        
        return out;
    }
};
